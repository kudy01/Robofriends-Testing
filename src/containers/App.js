import React, { useState, useEffect } from "react";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import Scroll1 from "../components/scroll1";
import ErrorBoundary from "../components/ErrorBoundary";
import "./App.css";

const App = () => {
  const [robots, setRobots] = useState([]);
  const [searchfield, setSearchfield] = useState("");

  useEffect(() => {
    fetch("https://jsonplaceholder.typicode.com/users")
      .then((response) => response.json())
      .then((users) => {
        setRobots(users);
      });
  }, []);

  const onSearchChange = (event) => {
    setSearchfield(event.target.value);
  };

  const filteredRobots = robots.filter((robot) => {
    return robot.name.toLowerCase().includes(searchfield.toLowerCase());
  });

  return !robots.length ? ( // when the page takes time to load as all the names come from an api
    <h1>Loading</h1>
  ) : (
    <div className="tc">
      <h1 className="f1">RoboFriends</h1>
      <SearchBox searchChange={onSearchChange} />
      <Scroll1>
        <ErrorBoundary>
          <CardList robots={filteredRobots} />
        </ErrorBoundary>
      </Scroll1>
    </div>
  );
};

export default App;
