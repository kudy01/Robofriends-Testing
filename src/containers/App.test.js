import React from "react";
import { shallow } from "enzyme";
import App from "./App";

it("Expect to render App Component", () => {
  expect(shallow(<App />)).toMatchSnapshot(); 
}); 
