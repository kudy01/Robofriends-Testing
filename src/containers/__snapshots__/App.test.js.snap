// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render App Component 1`] = `
<h1>
  Loading
</h1>
`;
